mlir/lib/IR/MLIRContext.cpp

```diff
@@ -711,6 +714,27 @@ ArrayRef<RegisteredOperationName> MLIRContext::getRegisteredOperations() {
   return impl->sortedRegisteredOperations;
 }
 
+/// Return information for registered operations by dialect.
+ArrayRef<RegisteredOperationName>
+MLIRContext::getRegisteredOperationsByDialect(StringRef dialectName) {
+  auto lowerBound =
+      std::lower_bound(impl->sortedRegisteredOperations.begin(),
+                       impl->sortedRegisteredOperations.end(), dialectName,
+                       [](auto &lhs, auto &rhs) {
+                         return lhs.getDialect().getNamespace().compare(
+                             rhs.getDialect().getNamespace());
+                       });
+
+  auto upperBound =
+      std::lower_bound(lowerBound, impl->sortedRegisteredOperations.end(),
+                       dialectName, [](auto &lhs, auto &rhs) {
+                         return lhs.getDialect().getNamespace().compare(
+                             rhs.getDialect().getNamespace());
+                       });
```

Comment:
You shouldn't need to compute the upper bound here, you should be able to use the operationCount that is computed in the dialect. That can be grabbed from the first operation of the lowerBound (after checking it's dialect is the same as the input, to account for no operations for the dialect).

Reply:
In the case of no operations is returning `.end()` fine?

Reply:
Actually, if we do this we don't need any extra count variable, so isn't this better?